Thanks to answers at
http://stackoverflow.com/questions/2407605/c-warning-missing-sentinel-in-function-call
for the hints on properly ensuring the sentinel argument.

--- a/src/zeroconf-reg.c	2016-04-27 23:31:19.000000000 +0200
+++ b/src/zeroconf-reg.c	2016-05-10 16:12:16.868101958 +0200
@@ -97,7 +97,7 @@
                     "gnuhost="GNU_HOST,
                     v ? version : NULL,
                     m ? machine : NULL,
-                    NULL) < 0) {
+                    (void*)NULL) < 0) {
             rs_log_crit("Failed to add service: %s\n", avahi_strerror(avahi_client_errno(ctx->client)));
             goto fail;
         }
